logic.js

```diff
@@ -0,0 +1,136 @@
+const Tweeter = function () {
+    let posts = [{
+            text: "First post!",
+            id: "p1",
+            comments: [{
+                    id: "c1",
+                    text: "First comment on first post!"
+                },
+                {
+                    id: "c2",
+                    text: "Second comment on first post!!"
+                },
+                {
+                    id: "c3",
+                    text: "Third comment on first post!!!"
+                }
+            ]
+        },
+        {
+            text: "Aw man, I wanted to be first",
+            id: "p2",
+            comments: [{
+                    id: "c4",
+                    text: "Don't wory second poster, you'll be first one day."
+                },
+                {
+                    id: "c5",
+                    text: "Yeah, believe in yourself!"
+                },
+                {
+                    id: "c6",
+                    text: "Haha second place what a joke."
+                }
+            ]
+        }
+    ]
+
+    let postIdCounter = 0,
+        commentIdCounter = 0
+
+    const getPosts = function () {
+        return posts
+    }
+
+    const addPost = function (text) {
+        postIdCounter = posts.length + 1
```

Comment:
The postIdCounter is not a counter for how many posts are there. but, rather an id generator. Therefore it's important to have a unique id. when assigning the counter to be like the array length, you may get the same id for two different posts